fineract-provider/src/main/java/org/apache/fineract/cob/service/InlineLoanCOBExecutorServiceImpl.java

```diff
@@ -94,6 +99,20 @@ public CommandProcessingResult executeInlineJob(JsonCommand command, String jobN
 
     @Override
     public void execute(List<Long> loanIds, String jobName) {
+        LocalDate oldestCOBBusinessDate = getOldestCOBBusinessDate(loanIds);
+        LocalDate cobBusinessDate = ThreadLocalContextUtil.getBusinessDateByType(BusinessDateType.COB_DATE);
```

Comment:
Aren't we missing a condition here to check whether the oldest date is the same as the current COB business date and if that's the case, just skip the execution? And probably we should do the same thing to remove loans which is in this state.
Might be wrong though, what do you think?
cc @adamsaghy

Reply:
tend to agree. what is the use case here? when will this service to be called?